homeassistant/components/luxtronik/binary_sensor.py

```diff
@@ -0,0 +1,127 @@
+"""Support for Luxtronik heatpump binary states."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import PLATFORM_SCHEMA, BinarySensorDevice
+from homeassistant.const import CONF_FRIENDLY_NAME, CONF_ICON, CONF_ID, CONF_SENSORS
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import slugify
+
+from . import DATA_LUXTRONIK, ENTITY_ID_FORMAT
+from .const import (
+    CONF_CALCULATIONS,
+    CONF_GROUP,
+    CONF_INVERT_STATE,
+    CONF_PARAMETERS,
+    CONF_VISIBILITIES,
+)
+
+ICON_ON = "mdi:check-circle-outline"
+ICON_OFF = "mdi:circle-outline"
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DEVICE_CLASS = "binary_sensor"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_SENSORS): vol.All(
+            cv.ensure_list,
+            [
+                {
+                    vol.Required(CONF_GROUP): vol.All(
+                        cv.string,
+                        vol.Any(CONF_PARAMETERS, CONF_CALCULATIONS, CONF_VISIBILITIES),
+                    ),
+                    vol.Required(CONF_ID): cv.string,
+                    vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+                    vol.Optional(CONF_ICON): cv.string,
+                    vol.Optional(CONF_INVERT_STATE, default=False): cv.boolean,
+                }
+            ],
+        )
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Luxtronik binary sensor."""
+    luxtronik = hass.data.get(DATA_LUXTRONIK)
+    if not luxtronik:
```

Comment:
This cannot happen. We don't set up platforms without setting up the integration .